sdk/core/src/seekable_stream.rs

```diff
@@ -0,0 +1,97 @@
+use async_std::io::{Read, Seek, SeekFrom};
+use async_std::prelude::*;
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::stream::Stream;
+use std::pin::Pin;
+
+#[async_trait]
+pub trait SeekableStream:
+    Read + Unpin + std::fmt::Debug + Send + Sync + dyn_clone::DynClone
```

Comment:
Conceptually, what is `DynClone` used for here? It is saying that the SeekableStream may be cloned?

Reply:
Yes, the stream must be clonable because reqwest (and I suppose any other client http implementation) takes ownership of it, but we need ownership back for subsequent retries. 

I've added a single function (`reset`) that replaces the `AsyncSeek` trait because we do not need arbitrary seeking, just to able to start from the beginning.

This is an abomination, I agree (also I do not know what could possibly implement it). In the previous iteration of the code I've implemented the trick of saving the original stream so we could replay the saved copy during retries. In other words, the original stream would only be executed once: our code would send the "saved" copy if a retry is needed. The trait than didn't need to be `clone`.